internal/aws/cwlogs/cwlog_client.go

```diff
@@ -42,23 +45,42 @@ const (
 type Client struct {
 	svc          cloudwatchlogsiface.CloudWatchLogsAPI
 	logRetention int64
+	tags         map[string]*string
+	accountID    string
+	region       string
 	logger       *zap.Logger
 }
 
 // Create a log client based on the actual cloudwatch logs client.
-func newCloudWatchLogClient(svc cloudwatchlogsiface.CloudWatchLogsAPI, logRetention int64, logger *zap.Logger) *Client {
+func newCloudWatchLogClient(svc cloudwatchlogsiface.CloudWatchLogsAPI, logRetention int64, tags map[string]*string, accountID string, region string, logger *zap.Logger) *Client {
 	logClient := &Client{svc: svc,
 		logRetention: logRetention,
+		tags:         tags,
+		accountID:    accountID,
+		region:       region,
 		logger:       logger}
 	return logClient
 }
 
 // NewClient create Client
-func NewClient(logger *zap.Logger, awsConfig *aws.Config, buildInfo component.BuildInfo, logGroupName string, logRetention int64, sess *session.Session) *Client {
+func NewClient(logger *zap.Logger, awsConfig *aws.Config, buildInfo component.BuildInfo, logGroupName string, logRetention int64, tags map[string]*string, sess *session.Session) *Client {
 	client := cloudwatchlogs.New(sess, awsConfig)
 	client.Handlers.Build.PushBackNamed(handler.RequestStructuredLogHandler)
 	client.Handlers.Build.PushFrontNamed(newCollectorUserAgentHandler(buildInfo, logGroupName))
-	return newCloudWatchLogClient(client, logRetention, logger)
+	// Assign region if it is defined in the config
+	region := ""
+	if !reflect.ValueOf(awsConfig.Region).IsNil() {
```

Comment:
Why is reflection used here?  Don't we know the type?  

```suggestion
	if awsConfig.Region != nil {
```

If there is concern that `awsConfig` may also be `nil` then check that first and return an error if it is as it is a required argument.

Reply:
I used reflection because when I was using `awsConfig.Region` it was giving me a null pointer exception.

Reply:
Although, now that I add the tags in `CreateLogGroup` there is no need for this section.